pkg/scheduler/internal/cache/cache.go

```diff
@@ -221,37 +223,76 @@ func (cache *schedulerCache) UpdateNodeInfoSnapshot(nodeSnapshot *nodeinfosnapsh
 			node.info.TransientInfo.ResetTransientSchedulerInfo()
 		}
 		if np := node.info.Node(); np != nil {
-			nodeSnapshot.NodeInfoMap[np.Name] = node.info.Clone()
+			existing, ok := nodeSnapshot.NodeInfoMap[np.Name]
+			if !ok {
+				nodesAdded = true
+				existing = schedulernodeinfo.NewNodeInfo()
+				nodeSnapshot.NodeInfoMap[np.Name] = existing
+			}
+			new := node.info.Clone()
+			if (len(existing.PodsWithAffinity()) > 0) != (len(new.PodsWithAffinity()) > 0) {
```

Comment:
Is "number of pods" enough to track? Won't it be problematic if one pod with affinity was added and a different previously existing was deleted?

Reply:
Never mind - I think I see now what it is used for.

Reply:
yeah, we just need to know whether there is a pod with affinity or not.